src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -105,6 +105,11 @@ Mass Segment::Solution::computeDeltaMass() const
     return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
 }
 
+Array<Maneuver> Segment::Solution::getManeuvers() const
+{
+    // Construct maneuvers from the contributions inside of a maneuvering segment
```

Comment:
Going to use the logic from dynamics_interpreter from the guidance service here to split thrusting accelerations produced during a maneuvering segment into multiple maneuvers

Reply:
On this
Whereas I agree that this is to be done
Maybe for this PR is good enough to simply return an error if there are more than 1 items?

Just saying as that particular implementation, whilst "easy on paper" will increase the scope and size of this PR

Reply:
Good point, will think about this